roles/init/tasks/postgres.yml

```diff
@@ -247,16 +247,37 @@
   with_dict: >-
     {{ postgres_tablespaces|default({}) }}
 
+# Determine the upstream primary for this host, that is, the primary at
+# the root of the cascading replication setup. This may be the instance
+# itself, if it's a primary.
+
+- name: Record name of the upstream primary
+  set_fact:
+    upstream_primary: "{{ inventory_hostname|upstream_root(hostvars) }}"
+  when: >
+    'postgres' in role
+
+- block:
+  - name: Prevent repmgr is installed on wrong BRD nodes when creating a replica
+    assert:
+      msg: >-
+        The upstream must be a subscriber-only node
+      that: "'subscriber-only' in hostvars[upstream_primary].role"
+    when: >
+      'replica' in role
```

Comment:
This should be limited to BDR nodes (which the other assert does).

Reply:
It's limited right? A few lines below, the entire block has 
`'bdr' in role`

Reply:
Yep, looks ok.